rust/interpreter/src/evaluate.rs

```diff
@@ -24,25 +27,43 @@ impl fmt::Display for EvalError {
 }
 
 trait Helpers {
+    fn read_u16(&self, offset: usize) -> u16;
     fn read_i32(&self, offset: usize) -> i32;
     fn read_offset(&self, offset: usize) -> isize;
+    fn read_atom(&self, offset: usize) -> String;
 }
 
 impl Helpers for EmitResult {
+    fn read_u16(&self, offset: usize) -> u16 {
+        u16::from_le_bytes(self.bytecode[offset..offset + 2].try_into().unwrap())
+    }
+
     fn read_i32(&self, offset: usize) -> i32 {
         i32::from_le_bytes(self.bytecode[offset..offset + 4].try_into().unwrap())
     }
 
     fn read_offset(&self, offset: usize) -> isize {
         self.read_i32(offset) as isize
     }
+
+    fn read_atom(&self, offset: usize) -> String {
+        self.strings[self.read_i32(offset) as usize].clone()
+    }
+}
+
+fn print(_: JSValue, args: Vec<JSValue>) -> JSValue {
```

Comment:
You could call the first argument something like `_this_value`. The leading underscore silences the Rust warning about the argument being unused.

Reply:
It's a bit more common in Rust to pass a slice reference, `args: &[JSValue]`.

Reply:
Fixed. I had to implement a custom `Debug` implementation for JSValue now, because slices can't be derived automatically.